<a href="https://colab.research.google.com/github/Kknorth51/HomeWork/blob/main/Stat_HW_7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Непараметрические тесты**

In [1]:
import numpy as np
import scipy.stats as stats

1-4 задачи решать с помощью функций, 5ю вручную
Выбрать тест и проверить, есть  ли различия между выборками:

1 )  Даны две  независимые выборки. Не соблюдается условие нормальности
x1  380,420, 290
y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

In [2]:
x1 = np.array([380,420, 290])
y1 = np.array([140,360,200,900])
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

В результате теста получили pvalue=0.629 > 0.05. Следовательно, у нас недостаточно оснований для отклонения нулевой гипотезы. Статистически значимых различий между выборками нет.

2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150,  130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

In [3]:
before = np.array([150, 160, 165, 145, 155])
after_10 = np.array([140, 155, 150, 130, 135])
after_30 = np.array([130, 130, 120, 130, 125])
stats.friedmanchisquare(before, after_10, after_30)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue=0.008 < 0.05 или pvalue=0.008 < 0.01. Следовательно, статистически значимые различия есть.

3 ) Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [4]:
stats.wilcoxon(before, after_10)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.0625 > 0.05 | 0.01. Следовательно, статистически значимых различий нет.

4) Даны 3 группы  учеников плавания.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [5]:
gr_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(gr_1, gr_2, gr_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue=0.065 > 0.05 | 0.01. Следовательно, статистически значимых различий между группами нет.

5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [6]:
sample = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
a = 0.05
M_0 = 2.5
M_1 = np.mean(sample)
std = np.std(sample, ddof=1)
n = 10
t = (M_1 - M_0) / (std / n ** 0.5)
t_table = stats.t.ppf(1 - a/2, n-1)

if abs(t) > t_table:
    print('Гипотеза отвергнута')
else:
    print('Гипотеза принята')
print(f'{t=}, {t_table=}')

Гипотеза принята
t=0.5630613661802959, t_table=2.2621571627409915
